synapse/rest/client/room.py

```diff
@@ -908,7 +908,7 @@ async def on_POST(
         return 200, return_value
 
     def _has_3pid_invite_keys(self, content: JsonDict) -> bool:
-        for key in {"id_server", "medium", "address"}:
+        for key in {"id_server", "medium", "address", "id_access_token"}:
```

Comment:
It is only used in line 863, and if any of these keys is not present it just does not pass further. Should it also return an error after this check?

Reply:
> if any of these keys is not present it just does not pass further

I don't think this is correct. If any of these keys is not present it treats the request as a regular invite rather than a 3pid invite.

I would say: if the request has `medium` and `address` keys, we should treat it as a 3pid invite. Then, if it does not have `id_server` or `id_access_token`, we should treat that as an error.

Reply:
This is addressed in latest update